conanfile.py

```diff
@@ -0,0 +1,97 @@
+import os
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.scm import Git, Version
+from conan.tools.files import update_conandata, download, unzip
+
+required_conan_version = ">=2.0"
+
+class DPPConan(ConanFile):
+    name = "dpp"
+    version = "10.0.33"
+    default_user = "brainboxdotcc"
+    default_channel = "testing"
+    license = "https://github.com/brainboxdotcc/DPP/blob/master/LICENSE"
+    url = "https://github.com/brainboxdotcc/DPP"
+    description = "D++ is a lightweight and efficient library for Discord"
+    topics = ("discord")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": True, "fPIC": True}
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
```

Comment:
This is very often limited mostly by the cppstd minimum, not the compiler version itself, and it was a way to achieve some checks in Conan 1. 

Conan 2 mandates/strongly recommend usage of ``cppstd`` in profiles, so checking the min_cppstd might be enough in many cases.